src/dealer/DealerWallet.ts

```diff
@@ -0,0 +1,515 @@
+import _ from "lodash"
+import { yamlConfig } from "../config"
+import { accountDealerFtxPath, liabilitiesDealerFtxPath } from "../ledger/ledger"
+import { MainBook } from "../mongodb"
+import { OnChainMixin } from "../OnChain"
+import { ILightningWalletUser } from "../types"
+import { btc2sat, sleep } from "../utils"
+import { baseLogger } from "../logger"
+import { UserWallet } from "../userWallet"
+import assert from "assert"
+
+import { GenericExchange, SupportedExchanges, ApiConfig } from "./GenericExchange"
+
+const activeExchangeId: SupportedExchanges = "ftx"
+
+const key = process.env[`${activeExchangeId.toUpperCase()}_KEY`]
+const secret = process.env[`${activeExchangeId.toUpperCase()}_SECRET`]
+const password = process.env[`${activeExchangeId.toUpperCase()}_PASSWORD`]
+
+if (!key || !secret || !password) {
+  throw new Error(`Missing ${activeExchangeId} exchange environment variables`)
+}
+
+const activeApiConfig = new ApiConfig(activeExchangeId, key, secret, password)
+
+// TODO: move to the yaml config
+const simulateOnly = true
+
+export type IBuyOrSell = "sell" | "buy" | null
```

Comment:
I don't like this being potentially `null`.
Can we have the type be `IBuyOrSell` and then where its consumed use `arg: IBuyOrSell | null`? Its better if the types we define have uniform behaviour and the `null` is called out locally at point of usage.

Reply:
yup this is code from me. it could be something like `"sell"| "buy" | "noAction"`?

Reply:
Do we need the `buy`/`sell` to be a string? Are we parsing this type from the return of an API call? If its something we control ourselves then I think an enum would be more appropriate:
```
export enum TradeAction {
  Buy,
  Sell,
  NoAction,
}
```

Reply:
- I will review the names to communicate more on a second pass, I was trying to alter the code to a minimum, so left the comments and todos also as is.
- try/catch was on my list also, we have some that just "safe guard" our own asserts but mostly they might be better place around the 3rd party api library
- I agree with the typed strings, there's a few occurrences, I will review them
- done with extra vars pre return...